src/LEGO.AsyncAPI.Readers/Exceptions/AsyncApiUnsupportedSpecVersionException.cs

```diff
@@ -11,26 +11,28 @@ namespace LEGO.AsyncAPI.Readers.Exceptions
     [Serializable]
     public class AsyncApiUnsupportedSpecVersionException : Exception
     {
-        const string MessagePattern = "AsyncApi specification version '{0}' is not supported.";
+        private const string MessagePattern = "AsyncApi specification version '{0}' is not supported.";
 
         /// <summary>
-        /// Initializes the <see cref="AsyncApiUnsupportedSpecVersionException"/> class with a specification version.
+        /// Initializes a new instance of the <see cref="AsyncApiUnsupportedSpecVersionException"/> class.
         /// </summary>
         /// <param name="specificationVersion">Version that caused this exception to be thrown.</param>
-        public AsyncApiUnsupportedSpecVersionException(string specificationVersion)
-            : base(string.Format(CultureInfo.InvariantCulture, MessagePattern, specificationVersion))
+        /// <param name="settings">The settings used for reading and writing.</param>
+        public AsyncApiUnsupportedSpecVersionException(string specificationVersion, AsyncApiSettings settings)
+            : base(string.Format(settings.CultureInfo, MessagePattern, specificationVersion))
         {
             this.SpecificationVersion = specificationVersion;
         }
 
         /// <summary>
-        /// Initializes the <see cref="AsyncApiUnsupportedSpecVersionException"/> class with a specification version and
+        /// Initializes a new instance of the <see cref="AsyncApiUnsupportedSpecVersionException"/> class.
         /// inner exception.
         /// </summary>
         /// <param name="specificationVersion">Version that caused this exception to be thrown.</param>
+        /// <param name="settings">The setting used for reading and writing</param>
         /// <param name="innerException">Inner exception that caused this exception to be thrown.</param>
-        public AsyncApiUnsupportedSpecVersionException(string specificationVersion, Exception innerException)
-            : base(string.Format(CultureInfo.InvariantCulture, MessagePattern, specificationVersion), innerException)
+        public AsyncApiUnsupportedSpecVersionException(string specificationVersion, AsyncApiSettings settings, Exception innerException)
+            : base(string.Format(settings.CultureInfo, MessagePattern, specificationVersion), innerException)
```

Comment:
to be honest, i skipped this intentionally.
I dont think pushing settings into an Exception like this makes sense.
And i think we can safely assume that exception formatting is up to us, the library authors.
Lets keep InvariantCulture here and not push settings down.